elytron-oidc-client/src/main/java/org/wildfly/extension/elytron/oidc/ElytronOidcSubsystemSchema.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.extension.elytron.oidc;
+
+
+import org.jboss.as.controller.Feature;
+import org.jboss.as.controller.PersistentSubsystemSchema;
+import org.jboss.as.controller.PersistentResourceXMLDescription;
+import org.jboss.as.controller.SubsystemSchema;
+import org.jboss.as.controller.xml.VersionedNamespace;
+import org.jboss.as.version.Stability;
+import org.jboss.staxmapper.IntVersion;
+
+import java.util.EnumSet;
+import java.util.Map;
+
+import static org.wildfly.extension.elytron.oidc.ElytronOidcSubsystemResourceRegistrar.SIMPLE_ATTRIBUTE_MARSHALLER;
+import static org.wildfly.extension.elytron.oidc.ElytronOidcSubsystemResourceRegistrar.SIMPLE_ATTRIBUTE_PARSER;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.ADAPTER_STATE_COOKIE_PATH;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.BEARER_ONLY;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.CLIENT_ID;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.ENABLE_BASIC_AUTH;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.MIN_TIME_BETWEEN_JWKS_REQUESTS;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.PUBLIC_CLIENT;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.PUBLIC_KEY_CACHE_TTL;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.RESOURCE;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.TOKEN_MINIMUM_TIME_TO_LIVE;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.TURN_OFF_CHANGE_SESSION_ID_ON_LOGIN;
+import static org.wildfly.extension.elytron.oidc.SecureDeploymentDefinition.USE_RESOURCE_ROLE_MAPPINGS;
+
+/**
+ * Enumerated the schema versions for the elytron-oidc-client subsystem.
+ * @author Prarthona Paul
+ */
+
+public enum ElytronOidcSubsystemSchema implements PersistentSubsystemSchema<ElytronOidcSubsystemSchema> {
+    VERSION_1_0(1),
+    VERSION_2_0(2),
+    VERSION_3_0_COMMUNITY(3, 0, Stability.COMMUNITY),
```

Comment:
Since there is no community-specific namespace for 2.0, you don't need to increment the version.
e.g.
```java
VERSION_2_0_COMMUNITY(2, 0, Stability.COMMUNITY)
```

Reply:
I have changed the community version to version 2 and updated the schema and test files. 
Quick question, when attributes in 2.0 community move to default, will they be part of subsystem version 3.0 or 2.0?

Reply:
Since 2.0 (default) already exists, you would increment to 3.0.

Reply:
So, would i go back to ` VERSION_3_0_COMMUNITY(3, 0, Stability.COMMUNITY),`?

Reply:
You would only introduce community:3.0 if you are adding new features to the existing community:2.0.

Reply:
@PrarthonaPaul Since there isn't a community-specific namespace for 2.0, you can use:

`VERSION_2_0_COMMUNITY(2, 0, Stability.COMMUNITY)`.

Later, if we were to add new features to the existing community:2.0, that's when we'd use `VERSION_3_0_COMMUNITY(3, 0, Stability.COMMUNITY)`. Or if we were to make your feature part of the default stability level instead, we'd introduce VERSION_3_0.

Reply:
Thanks @fjuma for clarifying. 
So, just to confirm, this means in the model versions we would just have version 2 and 1, no version 3. And the contents of from3(...) function (please see below) would be shifted to from2(...). Is that correct? 
https://github.com/wildfly/wildfly/blob/74260706d505c5df38e840d6ba58d09b309d54a6/elytron-oidc-client/src/main/java/org/wildfly/extension/elytron/oidc/ElytronOidcSubsystemTransformers.java#L47-L54

Reply:
The model version of a subsystem identifies a set of potential features exposed by a subsystem, independent of whether or not they are enabled (or indeed enablable) by a user's server configuration.  For a given version of WildFly, the ModelVersion of a subsystem is fixed.  This value is merely descriptive, and is largely invisible to users.  In general, a subsystem increments this version when the set of potential features has changed since the previous release.

That said, for this PR, you want to increment your subsystem model version to 3.0.0.

In contrast, a subsystem's XML schema describes what can be configured for a given subsystem.  A given version of WildFly can boot using XML that complies with any schema version of a given subsystem.  However, for a given subsystem, _what_ can be configured is constrained by the stability level of the server process.  That we version these using numbers is merely a convention (albeit a useful one), as is the inclusion of the stability level to the namespace uri - the uniqueness of the namespace URI is all that actually matters.

In general, we need to create a new schema version if the current schema version was already published by a previous WildFly release.

_Technically_, the only requirement for a new schema is that its namespace should be distinct from any existing one.  In other words, you _could_ name these however you wanted, so long as you take care to only register schemas supported by the current server process.  

That said, requiring subsystems to perform conditional schema registration depending on the stability level of the server is something we want to avoid.  To that effect, the ExtensionParsingContext, i.e. the thing with which you register your subsystem parsers, will introspect the Stability associated with a given SubsystemSchema and respond appropriately.

That said, for this PR, option 1 is clearly not an option, since it requires a new/distinct schema version.

If you used "wildfly:elytron-oicd-client:3.0" for the new namespace, the default logic of SubsystemSchema.getStability() would return Stability.DEFAULT.  This would otherwise cause ExtensionParsingContext to treat this as a valid parser for a server started using --stability=default, which would be incorrect.  Technically, you could override the value return by this method, but that would likely introduce confusion, as it would not be clear to which stability level the schema applies.

That leaves 2 obvious options:
* "wildfly:elytron-oicd-client:community:2.0"
* "wildfly:elytron-oicd:client:community:3.0"

Either will technically work with the existing infrastructure as they are distinct from the existing "wildfly:elytron-oicd-client:2.0".

Let's say the next release of WildFly adds a new preview feature to this subsystem.  For a server running in preview, the new schema includes changes from this PR.  Using the first option, the next schema becomes "wildfly:elytron-oicd-client:preview:2.0".  However, using the 2nd option, the next schema becomes "wildfly:elytron-oicd:client:preview:4.0".

Now consider a subsequent release of WildFly adding a new default feature to this subsystem.  We now have 3 schemas to manage, depending on the stability level of the server.
If we follow the conventions from the first option, the new names become: 
"wildfly:elytron-oicd:client:preview:3.0"
"wildfly:elytron-oicd:client:community:3.0"
"wildfly:elytron-oicd:client:3.0"
If we follow the conventions from the second option, the new names become:
"wildfly:elytron-oicd:client:community:3.0"

Reply:
Are you missing two namespaces in the last option @pferraro 

FYI I like "wildfly:elytron-oicd-client:community:2.0" as that seems to be what @Skyllarr did in her PR ;-)

Reply:
+1, I agree that going with 2.0 makes sense and matches what Diana did as well.

Reply:
Thanks @fjuma, @pferraro  and @darranl for your input. 
I have added a new commit to convert the schema to preview:2.0 since we have decided to move WFLY-17143 and WFLY-16532 to preview as well. I have also switched back to not incrementing the model version to 3.0.0.

Reply:
@PrarthonaPaul The model version still needs to be incremented to 3.0.